app/component/itinerary/navigator/NaviCardContainer.js

```diff
@@ -24,6 +24,26 @@ function addMessages(incominMessages, newMessages) {
     incominMessages.set(m.id, m);
   });
 }
+
+const handleLegChange = (leg, firstLeg, time) => {
+  let legType;
+  if (time < legTime(firstLeg.start)) {
+    legType = LEGTYPE.PENDING;
+  } else if (leg) {
+    if (!leg.transitLeg) {
+      if (leg.current >= TIME_AT_DESTINATION) {
+        legType = LEGTYPE.WAIT;
+      } else {
+        legType = LEGTYPE.MOVE;
+      }
+    } else {
+      legType = LEGTYPE.TRANSIT;
+    }
+  } else {
+    legType = LEGTYPE.WAIT;
+  }
+  return legType;
+};
```

Comment:
Well refactored! A pure function such as this can live it's merry, optimized life outside the component.